http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -872,68 +906,121 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
+    }
+    val landingZoneSupportsDatabase = landingZoneResources.postgresName.isDefined
+    if (databaseConfigEnabled && landingZoneSupportsDatabase) {
+      for {
+        // Build WSM client
+        auth <- samDao.getLeoAuthToken
+        token <- auth.credentials match {
+          case org.http4s.Credentials.Token(_, token) => F.pure(token)
+          case _ => F.raiseError(new RuntimeException("Could not obtain Leo auth token"))
+        }
+        wsmApi = wsmClientProvider.getControlledAzureResourceApi(token)
 
-        // Execute WSM call
-        createDatabaseResponse <- F.delay(wsmApi.createAzureDatabase(createDatabaseRequest, workspaceId.value))
-
-        _ <- logger.info(ctx.loggingCtx)(s"WSM create database response: ${createDatabaseResponse}")
-
-        // Poll for DB creation
-        // We don't actually care about the JobReport - just that it succeeded.
-        op = F.delay(wsmApi.getCreateAzureDatabaseResult(workspaceId.value, dbName))
-        result <- streamFUntilDone(
-          op,
-          config.appMonitorConfig.createApp.maxAttempts,
-          config.appMonitorConfig.createApp.interval
-        ).interruptAfter(config.appMonitorConfig.createApp.interruptAfter).compile.lastOrError
-
-        _ <- logger.info(ctx.loggingCtx)(s"WSM create database job result: ${result}")
-
-        _ <-
-          if (result.getJobReport.getStatus != JobReport.StatusEnum.SUCCEEDED) {
-            F.raiseError(
-              AppCreationException(
-                s"WSM database creation failed for app ${app.appName.value}. WSM response: ${result}",
-                Some(ctx.traceId)
-              )
-            )
-          } else F.unit
+        cromwellDb <- createDatabaseInWsm(app, workspaceId, namespace, "cromwell", wsmApi, None)
+        cbasDb <- createDatabaseInWsm(app, workspaceId, namespace, "cbas", wsmApi, None)
+        tesDb <- createDatabaseInWsm(app, workspaceId, namespace, "tes", wsmApi, None)
+      } yield Some(CromwellDatabaseNames(cromwellDb, cbasDb, tesDb))
+    } else F.pure(None)
+  }
 
-        // Save record in APP_CONTROLLED_RESOURCE table
-        _ <- appControlledResourceQuery
-          .save(app.id.id,
-                WsmControlledResourceId(result.getAzureDatabase.getMetadata.getResourceId),
-                WsmResourceType.AzureDatabase
+  private[util] def createDatabaseInWsm(app: App,
+                                        workspaceId: WorkspaceId,
+                                        namespace: KubernetesNamespace,
+                                        databaseNamePrefix: String,
+                                        wsmApi: ControlledAzureResourceApi,
+                                        owner: Option[UUID]
+  )(implicit
+    ev: Ask[F, AppContext]
+  ): F[String] = {
+    // Build create DB request
+    // Use the k8s namespace for the name. Note dashes aren't allowed.
+    val dbName = s"$databaseNamePrefix${namespace.name.value.split('-').head}"
```

Comment:
Just thinking out loud… I think underscores are allowed. `tes_z2okyl` might be a bit nicer than `tesz2okyl` for future us maintaining this.